core/src/banking_stage/consume_worker.rs

```diff
@@ -107,23 +107,26 @@ impl<Tx: TransactionWithMeta> ConsumeWorker<Tx> {
     fn consume(
         &self,
         bank: &Arc<Bank>,
-        work: ConsumeWork<Tx>,
+        mut work: ConsumeWork<Tx>,
     ) -> Result<(), ConsumeWorkerError<Tx>> {
         let output = self.consumer.process_and_record_aged_transactions(
             bank,
             &work.transactions,
             &work.max_ages,
         );
 
+        // Copy retryable indexes back to return message
+        work.retryable_indexes.extend(
+            output
+                .execute_and_commit_transactions_output
+                .retryable_transaction_indexes
+                .iter()
+                .cloned(),
+        );
```

Comment:
Copying the indexes here - possible to improve further by allowing the consumer to fill this in directly. But for now this PR just gets rid of the wonky allocation/deallocation pattern